commands/modmail.py

```diff
@@ -0,0 +1,161 @@
+
+        if modmail_content:
+            log.info('Message detected from user')
+            view = Confirm()
+            sent_message = await ctx.interaction.channel.send(
+                embed=gen_embed(title='Are you sure you want to send this?',
+                                content='Please verify the contents before confirming.'),
+                view=view)
+            timeout = await view.wait()
+            if timeout:
+                log.info('Confirmation view timed out')
+                for item in view.children:
+                    item.disabled = True
+                    await sent_message.edit(embed=gen_embed(title='Are you sure you want to send this?',
+                                                            content='Please verify the contents before confirming.'),
+                                            view=view)
+                    return
+            await sent_message.edit(embed=gen_embed(title='Are you sure you want to send this?',
+                                                    content='Please verify the contents before confirming.'),
+                                    view=view)
+
+            if view.value:
+                # if isinstance(ctx.channel, discord.TextChannel):
+                #     document = await db.servers.find_one({"server_id": ctx.guild.id})
+                #     role = None
+                #     if document['modrole']:
+                #         role = discord.utils.find(lambda r: r.id == document['modrole'], ctx.guild.roles)
+                #     permissions = ctx.channel.permissions_for(ctx.author)
+                #     if role:
+                #         if role not in ctx.author.roles:
+                #             if permissions.manage_messages is False:
+                #                 log.warning("Permission Error")
+                #                 await ctx.send(embed=gen_embed(title='Error',
+                #                                                content='Sorry, modmail does not work in regular text channels! Please use this command in a DM with me.'))
+                #                 log.warning("Error: modmail attempted to be sent from text channel")
+                #                 return
+                #     elif permissions.manage_messages is False:
+                #         log.warning("Permission Error")
+                #         await ctx.send(embed=gen_embed(title='Error',
+                #                                        content='Sorry, modmail does not work in regular text channels! Please use this command in a DM with me.'))
+                #         log.warning("Error: modmail attempted to be sent from text channel")
+                #         return
+
+                try:
+                    document = await db.servers.find_one({"server_id": ctx.guild.id})
+                except AttributeError:
+                    await ctx.send(embed=gen_embed(title='Modmail error',
```

Comment:
Best practice is to use ctx.respond, as it will auto determine if a followup message or interaction response should be sent.

```suggestion
                    await ctx.respond(embed=gen_embed(title='Modmail error',
```